doc/source/programs/gdal_vector_rasterize.rst

```diff
@@ -0,0 +1,214 @@
+
+    The band(s) to burn values into. May be repeated.
+
+.. option:: --invert
+
+    Invert rasterization. Burn the fixed burn value, or the burn value associated with the first feature into all parts of the image not inside the provided polygon.
+
+    .. note::
+
+        When the vector features contain a polygon nested within another polygon (like an island in a lake), GDAL must be built against GEOS to get correct results.
+
+.. option:: --all-touched
+
+    Enables the ALL_TOUCHED rasterization option.
+
+    .. note::
+
+       When this option is enabled, the order of the input features (lines or polygons) can affect the results. When two features touch each other, the last one (i.e. topmost) will determine the burned pixel value at the edge. You may wish to use the `--sql` option to reorder the features (ORDER BY) to achieve a more predictable result.
+
+.. option:: --burn <BURN>
+
+    Burn value. May be repeated.
+
+.. option:: -a, --attribute-name <ATTRIBUTE-NAME>
+
+    Attribute name.
+
+.. option:: --3d
+
+    Indicates that a burn value should be extracted from the "Z" values of the feature. Works with points and lines (linear interpolation along each segment). For polygons, works properly only if the are flat (same Z value for all vertices).
+
+.. option:: --add
+
+   Instead of burning a new value, this adds the new value to the existing raster. Suitable for heatmaps for instance.
+
+.. option:: -l, --layer, --layer-name <LAYER-NAME>
+
+    Indicates the layer(s) from the datasource that will be used for input features. May be specified multiple times, but at least one layer name or a -sql option must be specified (not both).
+
+.. option:: --where <WHERE>
+
+    An optional SQL WHERE style query expression to be applied to select features to burn in from the input layer(s).
+
+.. option:: --sql <SQL>|@<filename>
+
+    An SQL statement to be evaluated against the datasource to produce a virtual layer of features to be burned in.
+    Starting with GDAL 3.7, the @filename syntax can be used to indicate that the content is in the pointed filename.
+
+.. option:: --dialect <DIALECT>
+
+    SQL dialect. In some cases can be used to use (unoptimized) OGR SQL instead of the native SQL of an RDBMS by passing OGRSQL. The "SQLITE" dialect can also be used with any datasource.
+
+.. option:: --nodata <NODATA>
+
+        Assign a specified nodata value to output bands.
+
+.. option:: --init <INIT>
+
+    Pre-initialize output bands with specified value. May be repeated.
+
+.. option:: --srs <SRS>
+
+    Override the projection for the output file. If not specified, the projection of the input vector file will be used if available. When using this option, no reprojection of features from the SRS of the input vector to the specified SRS of the output raster, so use only this option to correct an invalid source SRS. The <srs_def> may be any of the usual GDAL/OGR forms, complete WKT, PROJ.4, EPSG:n or a file containing the WKT.
```

Comment:
SRS -> CRS also here.
English could probably be better My take:
The features are not reprojected from the CRS of the input vectors to the specified CRS of the output raster, so use this option only to correct an invalid source CRS.

Reply:
For the new gdal cli, we have standardized on "crs" for the official argument name, with "srs" as a hidden alias. The metavar and acronym should be CRS.

Reply:
Wouldn't be a good idea to create AddxxxxxArg for these commonly used options?
`--band` is also a good candidate, extent (or bbox) , resolution and size are other possible candidates.
An `.rst` include for each one would be also helpful for standardization.

Reply:
Indeed. gdal raster edit also has a "--crs"
AddBBOXArg() already exists. I was wondering if we should rather use it rather than the "--extent" you suggested